src/util/sourceMappingDecoder.js

```diff
@@ -116,4 +98,66 @@ function convertFromCharPosition (pos, lineBreakPositions) {
   }
 }
 
+function fromAstNode (astNode) {
+  if (astNode.src) {
+    var split = astNode.src.split(':')
+    return {
+      start: split[0],
+      length: split[1],
+      file: split[2]
+    }
+  }
+  return null
+}
+
+function findNodeAtInstructionIndex (astNodeType, instIndex, sourceMap, ast) {
+  var sourceLocation = atIndex(instIndex, sourceMap)
+  return findNodeAtSourceLocation(astNodeType, sourceLocation, ast)
+}
+
+function findNodeAtSourceLocation (astNodeType, sourceLocation, ast) {
+  var astWalker = new AstWalker()
+  var callback = {}
+  var found = null
+  callback[astNodeType] = function (node) {
+    var sourceMap = fromAstNode(node)
+    if (sourceMap.start < sourceLocation.start && sourceMap.start + sourceMap.length > sourceLocation.start + sourceLocation.length) {
```

Comment:
Return false if sourceLocation is not inside sourceMap's location.
Assign found only if the type matches astNodeType and then also return false;
else return true